src/reclat.c

```diff
@@ -1,434 +1,436 @@
+                    "(c) 2002-2004   Christian Borgelt"
+#define BLKSIZE  256            /* block size for enlarging vectors */
+
+
+/* --- output flags --- */
+#define OF_ABS        1         /* print absolute support */
+/* #define OF_SCANF      2         /\* convert names to scanable form *\/ */
+#define OF_LIST       4         /* list supporting transactions */
+
+/* --- target types --- */
+#define TT_SET        0         /* frequent item sets */
+#define TT_MFSET      1         /* maximally frequent item sets */
+#define TT_GRSET      2         /* generator item sets */
+#define TT_CLSET      3         /* closed item sets */
+#define TT_RULE       4         /* association rules */
+#define TT_HEDGE      5         /* association hyperedges */
+
+/* --- error codes --- */
+#define E_OPTION    (-5)        /* unknown option */
+#define E_OPTARG    (-6)        /* missing option argument */
+#define E_ARGCNT    (-7)        /* too few/many arguments */
+#define E_STDIN     (-8)        /* double assignment of stdin */
+#define E_TARGET    (-9)        /* invalid target type */
+#define E_SUPP     (-10)        /* invalid item set support */
+#define E_ITEMCNT  (-11)        /* invalid number of items */
+#define E_NOTAS    (-12)        /* no items or transactions */
+#define E_UNKNOWN  (-18)        /* unknown error */
+
+#define SEC_SINCE(t)  ((clock()-(t)) /(double)CLOCKS_PER_SEC)
+#define RECCNT(s)     (tfs_reccnt(is_tfscan(s)) \
+                      + ((tfs_delim(is_tfscan(s)) == TFS_REC) ? 0 : 1))
+#define BUFFER(s)     tfs_buf(is_tfscan(s))
+
+/*----------------------------------------------------------------------
+  Constants
+----------------------------------------------------------------------*/
+/* --- error messages --- */
+static const char *errmsgs[] = {
+  /* E_NONE      0 */  "no error\n",
+  /* E_NOMEM    -1 */  "not enough memory\n",
+  /* E_FOPEN    -2 */  "cannot open file %s\n",
+  /* E_FREAD    -3 */  "read error on file %s\n",
+  /* E_FWRITE   -4 */  "write error on file %s\n",
+  /* E_OPTION   -5 */  "unknown option -%c\n",
+  /* E_OPTARG   -6 */  "missing option argument\n",
+  /* E_ARGCNT   -7 */  "wrong number of arguments\n",
+  /* E_STDIN    -8 */  "double assignment of standard input\n",
+  /* E_TARGET   -9 */  "invalid target type '%c'\n",
+  /* E_SUPP    -10 */  "invalid minimal support %g\n",
+  /* E_ITEMCNT -11 */  "invalid number of items %d\n",
+  /* E_NOTAS   -12 */  "no items or transactions to work on\n",
+  /*    -13 to -15 */  NULL, NULL, NULL,
+  /* E_ITEMEXP -16 */  "file %s, record %d: item expected\n",
+  /* E_DUPITEM -17 */  "file %s, record %d: duplicate item %s\n",
+  /* E_UNKNOWN -18 */  "unknown error\n"
+};
+
+static const char* BMtarget[] = {
+  /* BM_NORMAL      0 */  "frequent itemsets",
+  /* BM_CLOSED      1 */  "closed frequent itemsets",
+  /* BM_MAXIMAL     2 */  "maximally frequent itemsets",
+  /* BM_GENERATOR   3 */  "generator frequent itemsets",

```

Comment:
/* Makh2018: Define "generator frequent itemsets" as  a constant target type for generator item sets*/